src/Database/Schema/Collection.php

```diff
@@ -86,10 +86,10 @@ public function listTables(): array
      *
      * @param string $name The name of the table to describe.
      * @param array $options The options to use, see above.
-     * @return \Cake\Database\Schema\TableSchemaInterface Object with column metadata.
+     * @return \Cake\Database\Schema\TableSchema Object with column metadata.
      * @throws \Cake\Database\Exception when table cannot be described.
      */
-    public function describe(string $name, array $options = []): TableSchemaInterface
+    public function describe(string $name, array $options = []): TableSchema
```

Comment:
We are going in the opposite direction by replacing the interface with concrete class. I would keep the interfacce typehint and add inline annotation if possible or simply ignore the error in phpstan's config file.